llvm/utils/TableGen/IntrinsicEmitter.cpp

```diff
@@ -155,13 +158,13 @@ void IntrinsicEmitter::EmitEnumInfo(const CodeGenIntrinsicTable &Ints,
 
   OS << "// Enum values for intrinsics.\n";
   bool First = true;
-  for (const auto &Int : Ints[*Set]) {
+  for (const auto &Int : Set->getIntrinsics()) {
     OS << "    " << Int.EnumName;
 
     // Assign a value to the first intrinsic in this target set so that all
     // intrinsic ids are distinct.
     if (First) {
-      OS << " = " << Set->Offset + 1;
+      Intrinsic::PrintIntrinsicIDEncoding(OS, Set->TargetIndex, 0);
```

Comment:
The 9 here will be stable independent on which targets are enabled or not. It's basically the index among all possible target prefixes defined in a particular LLVM build. That means that if someone added a new target prefix for say a new target, then its index would be susceptible to change.

I propose the following: get this working first, so the IDs are more stable than before and for the same LLVM build, they will stay the same irrespective of which targets are enabled. As a follow on, we add a new record to the Intrinsics.td file that assigns stable target ID to each target. The record can either be in Intrinsidc.td or in the individual per-backend TD files. Something like:

```
class TargetPrefixIndex<string Name, int Index> {
...
}

and then Intrinsics.td can have a table of these:

```
def x86_target_index : TargetPrefixIndex<"x86", 0>;
def r600_target_index : TargetPrefixIndex<"r600", 8>;
```
and then the IntrinsicEmitter will use that as opposed to the index computed by sorting the present target prefixes.
Let me know if that works, or you prefer doing that change in the current PR directly.

Reply:
Also, note that on discourse the desire to have stable IDs has been questioned, so it would be best to resolve that discussion first before adding more features in service of stable intrinsic IDs.

Reply:
Also, I don't know how much value the `TargetPrefixIndex` provides. Across different LLVM source versions (which is when a new target prefix will appear), intrinsic IDs won't be stable in any case even if the TargetPrefixIndex stays stable, since intrinsics themselves for a given target may have changed, causing a change in the intrinsic index that's encoded in the enum. To be truly stable, we need to be able to specify in TableGen both the index of the target prefix as well as the index of the intrinsic within that target (i.e, change the `Intrinsic` class to be able to specify it) and then add checks for duplicates, which seems quite invasive change.

Reply:
LLVM only guarantees API stability within major versions. This means we have to avoid back porting changes that add new intrinsics to release branches, or if we want the ability to do that, we'd need to assign every intrinsic a stable id. I'm not proposing that and I don't want to do it.